runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -398,7 +396,10 @@ public void processWatermark(Watermark mark) throws Exception {
         sdkHarnessRunner.setBundleFinishedCallback(
             () -> {
               try {
-                processWatermark(mark);
+                LOG.debug("processing pushed back watermark: {}", mark);
+                // at this point the bundle is finished, allow the watermark to pass
+                setPushedBackWatermark(mark.getTimestamp());
```

Comment:
Shouldn't we restore the original watermark hold here again?

Reply:
What if there was already a hold > mark?

Reply:
Changed to restore of previous hold.